app/models/user.rb

```diff
@@ -0,0 +1,41 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+
+  validates :fullname, :slug, :email, :role, :password, presence: true
+
+  ADMIN = 1
+  AUTHOR = 2
+
+  before_validation(on: :create) do
```

Comment:
There's a better way of writing hooks. You can see it here:  https://github.com/bbatsov/rails-style-guide#activerecord